types/router.go

```diff
@@ -35,18 +35,26 @@ type Route struct {
 	handler Handler
 }
 
-func NewRoute(p string, h Handler) *Route {
-	return &Route{path: p, handler: h}
+// NewRoute returns an instance of Route.
+func NewRoute(p string, h Handler) Route {
+	return Route{path: p, handler: h}
 }
 
+// Path returns the path the route has assigned.
 func (r Route) Path() string {
 	return r.path
 }
 
+// Handler returns the handler that handles the route.
 func (r Route) Handler() Handler {
 	return r.handler
 }
 
+// Nil returns true only if both handler and path are not empty.
+func (r Route) Nil() bool {
```

Comment:
```suggestion
func (r Route) Empty() bool {
```

I'd favor `Empty` over `IsEmpty` (same method naming we use for types in `types`

Reply:
Second this.